### ISSUE TITLE
Is NE (New Executable, win16) support is planned?

### ISSUE THREAD

**Issue by grrk-bzzt** [radare/radare2#2730]
Hello,

Everything is in the title. I was wondering if r2 could support NE at some point or if it's too complicated.
I've read some articles saying that NE executables were quite complex to disassemble.
https://github.com/radare/radare2-regressions/tree/master/bins/ne

**Comment by radare** [comment]
I'm open to accept new rbin plugins. But nobody proposed to support it yet, and as its not prioritary and necessary for my needs i will not add it to my todo. But i'll happily accept a pullreq.

I also want to support PE resources and .NET bins too, so the windows platform needs a bit more of attention.

**Comment by XVilka** [comment]
@radare PE bug is here: https://github.com/radare/radare2/issues/921

**Comment by Maijin** [comment]
@grrk-bzzt ; @marespiaut Can you add binary example for this issue (shareable == no copyright, no malware etc.) ? and same https://github.com/radare/radare2/issues/314 ? in https://github.com/radare/radare2-regressions/tree/master/bins

Thanks.

**Comment by marespiaut** [comment]
@Maijin Off the top of my head, I can give you a link to SkiFree 1.0 (https://ski.ihoc.net/ski.zip) which is downloadable from the official website (https://ski.ihoc.net/).

But I can find some shareware games or demos if it's not usable or if it's not enough.

**Comment by Maijin** [comment]
As long as it is no copyright (so no shareware obviously). 
Can you also add one LE here #314 ?

**Comment by marespiaut** [comment]
Maybe this software can be considered as freeware: https://archive.org/details/win3_ANIM8

The about section is the following:
```
Anim8
Version 1.02
(C) Copyright 1992
Eric Bergman-Terrell
All Right Reserved

This program may be freely distributed
without cost, provided it is not changed
in any way. If you find this program
useful, please send $5.00 to:

Pocket-Sized Software
8547 E. Arapahoe Road, Suite J-147
Greenwood Village, CO 80112 USA
```

**Comment by marespiaut** [comment]
In the worst case, I have some documentation at home to write Windows 3.1 software in C. It may be a good excuses to create some custom Windows 3.1x binaries for radare2.

**Comment by Maijin** [comment]
Post LE file in LE issue

**Comment by marespiaut** [comment]
> Post LE file in LE issue

@Maijin I don't understand what you are talking about. If LE stands for "OS/2 Linear Executable", I can assure you I have never seen or used one, nor I ever used anything OS/2 related. If it's the case, I'm unaware of it.

I only know about NE "New Executable" files.

**Comment by Maijin** [comment]
okok

**Comment by j123123** [comment]
Getting binary samples is not a problem. You can take free OpenWatcom compiler and generate as many 16-bit NE samples as you need. OpenWatcom also have debugger, which can handle this format

**Comment by Freakqency** [comment]
I have been developing a plugin for ne format. I was able to obtain the vaddr value by passing the values stored in cs and ip to a function to convert it to a 20bit address format .can anyone help me out with finding the paddr value.
I tried to replicate this function
``` 
static ut64 r_bin_mz_la_to_pa(const struct r_biin_mz_obj_t *bin, ut64 la ){
	return la + (bin->dos_header->header_paragraphs << 4);
}
```
but i dont know what to replace the header_paragraphs with

**Comment by mathiasbockwoldt** [comment]
I don't feel competent enough to help myself, but maybe this project could help in boosting NE support:
https://github.com/zfigura/semblance
"Disassembler for Windows executables. Supports 16-bit NE (New Executable), MZ (DOS), and PE (Portable Executable, i.e. Win32) files."

@Freakqency: you might want to have a look to see how zfigura did things.

**Comment by Maijin** [comment]
Finally Fixed ! @GustavoLCR feel free to check #314 which should be similar (LE Files)
